### PR TITLE
fix(devtools): add reconnect when websocket closed by server

### PR DESCRIPTION
**Pull request by Cyunong**
Before submitting a new pull request, please make sure:

- [ ] Test cases have been added/updated/passed for the code you will submit.
- [ ] Documentation has added or updated.
- [ ] Commit message is following the [Convention Commit](https://conventionalcommits.org/) guideline with maximum 72 characters.
- [ ] Squash the repeat code commits, short patches are welcome.

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/Hippy?pullRequest=4129) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/Hippy?pullRequest=4129) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/Hippy?pullRequest=4129) it.</sub>

**Review by wwwcg — DISMISSED** [review]
lgtm

File: devtools/devtools-integration/native/src/devtools_data_source.cc [hunk 1]
```diff
@@ -43,7 +43,7 @@ using StringViewUtils = footstone::stringview::StringViewUtils;
 static std::atomic<uint32_t> global_devtools_data_key{1};
 footstone::utils::PersistentObjectMap<uint32_t, std::shared_ptr<DevtoolsDataSource>> devtools_data_map;
 
-DevtoolsDataSource::DevtoolsDataSource(const std::string& ws_url,
+void DevtoolsDataSource::create_devtools_service(const std::string& ws_url,
```
Comment (zealotchen0) [inline devtools/devtools-integration/native/src/devtools_data_source.cc:46]: 这里方法名首字母大写，参考 c++ 规范，和现有代码保持一致

File: devtools/devtools-backend/src/tunnel/ws/web_socket_channel.cc [hunk 2]
```diff
@@ -153,14 +161,32 @@ void WebSocketChannel::HandleSocketConnectMessage(const websocketpp::connection_
...
+    attempt_reconnect();
+  }
+  else {
+    // set handle nullptr when connect fail
+    data_handler_ = nullptr;
+    unset_messages_.clear();
+  }
 }
+
+void WebSocketChannel::attempt_reconnect() {
```
Comment (zealotchen0) [inline devtools/devtools-backend/src/tunnel/ws/web_socket_channel.cc:180]: 方法名应该大写

**Review by zealotchen0 — DISMISSED** [review]

**Review by wwwcg — APPROVED** [review]

**Review by zealotchen0 — APPROVED** [review]

**Pull request merged**